### DISCUSSION TITLE
How can I avoid copies of the form filename-computername.ext?

### DISCUSSION THREAD

**Discussion by JackNWhite** [abraunegg/onedrive#2133] [Q&A] [answered]
Hi folks,

I am getting copies of files appearing in my Sharepoint shared library. The original is `filenname.ext` and the copy is `filename-mycomputername.ext`.  Is there any way to prevent this from happening?

**Comment by abraunegg** [comment]
@JackNWhite 
What application version are you using?

What is the output of `onedrive --version` ?

**Reply by Weissrolf** [reply]
I also see these from time to time and assume them to be synchronization conflicts. Cloud Sync on Synology does the same, adding "Diskstation" and "conflict" and a date/time to the name, the latter of which clarifies the situation much better than the Linux client's current approach of just adding the host name.

The Microsoft Onedrive client on Windows also creates similar files in case of conflicts. This happens especially when I turn on my Windows laptop every once in a while and stupidly start some software accessing Onedrive files before the sync process completed (due to the long time the laptop wasn't connecting).

In all of my own cases the copy using the original name usually is the latest version of the file while the copies with appendices in their names are older copies.

I think it mostly happens when a file is repeatedly modified (like database files) so that the local version is already changed again before its last state finished uploading.

onedrive v2.4.20-1+np1

**Reply by JackNWhite** [reply]
Hi, my version is `onedrive v2.4.20-1+np1`.

I guessed it was a syncing problem, but I hoped there was something I could do to avoid them.

**Comment by abraunegg** [comment, marked as answer]
@Weissrolf , @JackNWhite 

A conflict file (the file with the hostname) is generated in the following scenario:

1. The remote file has a different size or hash or is 'newer' based on file timestamp
2. A local file exists with the same name in the same location
3. The application has no record of the local file being in-sync

Now - for point 3, if you are using `--resync` *at all*, this will remove the local cache state, thus, the application has no record of what is in sync, thus, to preserve data and avoid potential data loss, creates the 'backup' of the local data to the the file with the hostname.

> hoped there was something I could do to avoid them

You can do the following to avoid those files from being created:
1. Do not use `--resync` unless the application advises you to because of configuration change
2. Use the 'config' file option `bypass_data_preservation = "true"` which will bypass the creation of these backup files - but then you are open to potential data loss scenarios .... use with caution.

**Reply by JackNWhite** [reply]
Sorry for the delay in my response.

Thanks for the info, but I am still confused about how these files are arising. Sometimes they occur for files I have not even opened, let alone changed. I am concerned by scenario 1 in your answer. Surely any file which has been updated by someone else will fulfil those criteria.

Case in point: today I think someone was working on a PowerPoint presentation directly on Sharepoint today when my daemon synced. The presentation, which I have never opened, was copied mid-edit. My reasoning here is that the file size in the hostname copy does not appear anywhere in the version history of the original document.

**Reply by abraunegg** [reply]
@JackNWhite 
I suspect you are not understanding how this client operates which is why you are confused. If you run the client in verbose mode (`--verbose`) this may give you some insight as to what is actually going on for you.

In your recent example, your file was opened and edited by someone else. Your file also resides on SharePoint. SharePoint 'further modifies' the file (it is a 'feature', not a bug according to Microsoft - read https://github.com/OneDrive/onedrive-api-docs/issues/935).

The client, on next sync, gets a notification that file X (the one that someone else modified) has changed. It needs to update your local copy. Now - because of https://github.com/OneDrive/onedrive-api-docs/issues/935 and all the 'enrichment' the file also has other 'different' tag items - which are no longer relevant to your local copy - so no referential integrity with your local data.

As the file the client needs to update, exists locally, this file is then 'backed up locally' by making existing local file have the computer name added to the file name, so that the new file can be downloaded.

> The presentation, which I have never opened, was copied mid-edit.

If you were editing this file, then, if the application you were using to edit the file had put a lock on the file, the rename of the file would have failed, thus the download would have also failed ...... so potentially you need to go back and look at your environment a little closer to understand what is going on.

SharePoint with its oddities is annoying - because of their 'feature' ........ 

To avoid the renaming, even more so when using SharePoint, you need to use `bypass_data_preservation = "true"` .. - but then you are open to potential data loss scenarios .... use with caution.

**Comment by Weissrolf** [comment]
Then in my case the files were likely created by --resync. No big problem, as I can just search for them and delete them. It's always good to have a data preservation mechanism in place.
